### PR TITLE
ice: add dependencies, fix Linux build

### PR DESCRIPTION
**Pull request by nandahkrishna**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

This PR attempts to fix `ice`'s build step to bottle it on Linux. I've bumped the revision since the changes also affect macOS.

### CODE REVIEW

File: Formula/ice.rb [hunk 1]
```diff
@@ -19,8 +20,11 @@ class Ice < Formula
   end
 
   depends_on "lmdb"
-  depends_on macos: :catalina
   depends_on "mcpp"
+  depends_on "openssl@3"
+
+  uses_from_macos "bzip2"
+  uses_from_macos "expat"
```
Comment (carlocab) [inline Formula/ice.rb:27]:
```suggestion
  uses_from_macos "expat"
  uses_from_macos "libedit"
```

**Comment by carlocab** [comment]
Let's document why we're moving stuff to `libexec` so other people reading the formula don't have to go spelunking: https://github.com/Homebrew/homebrew-core/pull/36783

**Review by SMillerDev — DISMISSED** [review]

File: Formula/ice.rb [hunk 2]
```diff
@@ -19,8 +20,12 @@ class Ice < Formula
   end
 
   depends_on "lmdb"
-  depends_on macos: :catalina
   depends_on "mcpp"
+  depends_on "openssl@3"
```
Comment (carlocab) [inline Formula/ice.rb:24]: No linkage with `openssl@3` on macOS. I think this is a Linux-only dependency. We can probably drop the revision bump.

**Review by carlocab — APPROVED** [review]

**Comment by carlocab** [comment]
macOS takes about an hour to build, but does so in under ten minutes on Linux. Suspicious...

**Pull request merged**